### PR TITLE
Cooldown mechanism - allow atlas to resolve challenges that remain unresolved after a timeout period

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: contracts/Front/Challenges.sol [hunk 1]
```diff
@@ -233,6 +233,10 @@ contract Challenges is Base {
 
     function isOnCooldown(address resolverId, bytes32 challengeId) private view returns (bool) {
         uint lastResolvedSequenceNumber = atlasStakeStore.getLastChallengeResolvedSequenceNumber(resolverId);
-        return lastResolvedSequenceNumber != 0 && lastResolvedSequenceNumber.add(getCooldown()) > getChallengeSequenceNumber(challengeId);
+
+        // solium-disable-next-line operator-whitespace
+        return getChallengeCreationTime(challengeId).add(config.COOLDOWN_TIMEOUT()) > time.currentTimestamp() &&
+        lastResolvedSequenceNumber != 0 &&
+        lastResolvedSequenceNumber.add(getCooldown()) > getChallengeSequenceNumber(challengeId);
```
Comment (Anze1m) [inline contracts/Front/Challenges.sol]: Jeeeeez, why do we have to put all these in one line? 😝 
```
if(timedOut) {
  return something;
}
if(sequenceNumberStuff) {
  return something;
}
return something;
```

File: contracts/Configuration/Config.sol [hunk 2]
```diff
@@ -23,6 +23,8 @@ contract Config {
 
     uint32 constant public COOLDOWN_AVAILABLE_TARGET_PERCENT = 90;
 
+    uint32 constant public COOLDOWN_TIMEOUT = 1 days;
```
Comment (Anze1m) [inline contracts/Configuration/Config.sol]: I'd rather like `24 hours` here

File: contracts/Front/Challenges.sol (hunk 1 above)
Comment (spherefoundry) [reply contracts/Front/Challenges.sol]: I would normally agree with @Anze1m, but in this case I don't. The return value is of type bool. So in the example above `something` would be `true` for all the if's, and false on the last. 

Instead, I would argue there should be a second method `challengeCooldownExpired` and both of them should be referenced when checking if a Atlas can resolve.

File: test/contracts/front/challenges.js [hunk 3]
```diff
@@ -408,6 +411,17 @@ describe('Challenges Contract', () => {
       await expect(resolveChallenge(challengeId, resolver)).to.be.eventually.fulfilled;
     });
 
+    it('Can resolve if resolver is on cooldown but challenge is older than cooldown timeout', async () => {
```
Comment (spherefoundry) [inline test/contracts/front/challenges.js]: `Can resolve ignoring resolver cooldown if challenge is older than cooldown timeout`

**Pull request merged**